firedrake/assemble.py

```diff
@@ -995,6 +1019,15 @@ def _apply_bcs_mat_real_block(op2tensor, i, j, component, node_set):
             dat = op2.DatView(dat, component)
         dat.zero(subset=node_set)
 
+    @cached_property
+    def parloop_builders_for_sparsity_construction(self):
```

Comment:
I think that this would definitely need documenting. What magic are we doing here?

Reply:
Fixed.